docs/HowToFAQ/Errors-Parser.md

```diff
@@ -0,0 +1,1194 @@
+<!-- %check-resolve %default -->
+
+<!-- Parser.cs, but not Deprecated warnings or syntactic errors -->
+
+## **Error: Duplicate declaration modifier: abstract**
+
+```dafny
+abstract abstract module M {}
+```
+
+No Dafny modifier, such as [`abstract`, `static`, `ghost`](../DafnyRef/DafnyRef#sec-declaration-modifiers) may be repeated. Such repetition would be superfluous even if allowed.
+
+## **Error: a _decl_ cannot be declared 'abstract'**
+ 
+```dafny
+module M {
+  abstract const c := 4
+}
+```
+
+Only some kinds of declarations can be declared abstract. 
+In the example, a const declaration cannot be abstract.
+
+## **Error: a function-by-method has a ghost function body and a non-ghost method body; a function-by-method declaration does not use the 'ghost' keyword.**
+
+```dafny
+ghost function f(): int
+{
+  42
+} by method {
+  return 42;
+}
+```
+
+Functions with a [by method](../DafnyRef/DafnyRef#sec-function-declarations)
+section to their body can be used both in ghost contexts and in non-ghost contexts; in ghost contexts the function body is used and in compiled contexts
+the by-method body is used. The `ghost` keyword is not permitted on the 
+declaration.
+
+## **Error: _decl_ cannot be declared 'ghost' (it is 'ghost' by default)**
+
+```dafny
+ghost function f(): int { 42 }
+```
+
+For versions prior to Dafny 4, the `function` keyword meant a ghost function
+and `function method` meant a non-ghost function. 
+From Dafny 4 on, `ghost function` means a ghost function and 
+`function` means a non-ghost function. 
+See the discussion [here](../DafnyRef/DafnyRef#sec-function-syntax) for
+a discussion of options to control this feature.
+
+## **Error: a _decl_ cannot be declared 'ghost'**
+
+```dafny
+ghost module M {}
+```
+
+Only some kinds of declarations can be declared `ghost`, most often functions,
+fields, and local declarations. In the example, a `module` may not be `ghost`.
```

Comment:
It would be great to have a word about it being either a current limitation of Dafny, or something that is not possible for X soundness reason. I don't understand why a module may not be ghost for example, we ought to say something like "as of today, a module may not be ghost"

Reply:
(I'm just thinking out loud.) A module today can be `abstract`. That means that the compiler will not generate any code for it. That sounds kind of like `ghost`. Except, if a module were declared as `ghost`, then I would expect that its contents would _never_ be compiled, even if a different module would refine this one. In contrast, when an `abstract` module is refined, its contents will be compiled (except contents that's declared as `ghost`, of course). So, my thinking-out-loud leads me to the conclusion: We should not replace `abstract module` by `ghost module`; instead, if we ever consider having a `ghost module`, then that's a different thing than an `abstract module`.

Reply:
And I try to avoid saying 'as of today' in a reference manual unless it is about something that I know is changing. Everything is as of today.

Reply:
I like Rustan's line of thought, but also understand that adding it to the documentation might be a bit too much here.